src/main/webapp/app/assessment/athena.service.ts

```diff
@@ -14,42 +18,119 @@ export class AthenaService {
+                        return feedback;
+                    }
+                    // Referenced feedback, convert to TextBlockRef
+                    feedback.type = FeedbackType.MANUAL;
+                    const textBlock = new TextBlock();
+                    textBlock.startIndex = suggestion.indexStart;
+                    textBlock.endIndex = suggestion.indexEnd;
+                    textBlock.setTextFromSubmission(submission);
+                    feedback.reference = textBlock.id;
+                    return new TextBlockRef(textBlock, feedback);
+                });
+            }),
+        );
+    }
+
+    /**
+     * Get feedback suggestions for the given programming submission from Athena
      *
-     * @param exerciseId the id of the exercise
+     * @param exercise
      * @param submissionId the id of the submission
+     * @return observable that emits the feedback suggestions as Feedback objects with the "FeedbackSuggestion:" prefix
      */
-    getFeedbackSuggestionsForText(exerciseId: number, submissionId: number): Observable<TextBlockRef[]> {
-        return this.profileService.getProfileInfo().pipe(
-            switchMap((profileInfo) => {
-                if (!profileInfo.activeProfiles.includes('athena')) {
-                    return of([] as TextBlockRef[]);
-                }
-                return this.http
-                    .get<TextBlockRef[]>(`${this.resourceUrl}/exercises/${exerciseId}/submissions/${submissionId}/feedback-suggestions`, { observe: 'response' })
-                    .pipe(switchMap((res: HttpResponse<TextBlockRef[]>) => of(res.body!)))
-                    .pipe(
-                        switchMap((feedbackSuggestions) => {
-                            // Make real TextBlockRef objects out of the plain objects
-                            return of(feedbackSuggestions.map((feedbackSuggestion) => new TextBlockRef(feedbackSuggestion.block!, feedbackSuggestion.feedback)));
-                        }),
-                    );
+    public getProgrammingFeedbackSuggestions(exercise: Exercise, submissionId: number): Observable<Feedback[]> {
+        return this.getFeedbackSuggestions<ProgrammingFeedbackSuggestion>(exercise, submissionId).pipe(
+            map((suggestions) => {
+                return suggestions.map((suggestion) => {
+                    const feedback = new Feedback();
+                    feedback.credits = suggestion.credits;
+                    feedback.text = FEEDBACK_SUGGESTION_IDENTIFIER + suggestion.title;
+                    feedback.detailText = suggestion.description;
+                    if (suggestion.filePath != undefined && suggestion.lineStart != undefined) {
+                        // Referenced feedback
+                        feedback.type = FeedbackType.MANUAL;
+                        feedback.reference = `file:${suggestion.filePath}_line:${suggestion.lineStart}`; // Ignore lineEnd for now because Artemis does not support it
```

Comment:
Wouldn't it make sense to use `lineEnd` instead of `lineStart`? I think the placement will be better then, both should be set (?)

Reply:
I changed it 👍